src/project/solve_group.rs

```diff
@@ -67,15 +67,10 @@ impl<'p> SolveGroup<'p> {
 
     /// Returns all features that are part of the solve group.
     ///
-    /// If `include_default` is `true` the default feature is also included.
-    ///
     /// All features of all environments are combined and deduplicated.
-    pub fn features(
-        &self,
-        include_default: bool,
-    ) -> impl DoubleEndedIterator<Item = &'p manifest::Feature> + 'p {
+    pub fn features(&self) -> impl DoubleEndedIterator<Item = &'p manifest::Feature> + 'p {
```

Comment:
Need to handle partial default feature here. we need to change all functions calling features to either:

- iter on environments instead of features
- add some logic to determine if default feature should be included or not

Reply:
I don't fully understand what is required here. Where in the code do you need to have some extra logic on the features where you need to get the default logic from the environments?

Reply:
@ruben-arts for most methods of SolveGroup that are currently calling features() directly

Reply:
I kept iterating on features, including default feature if any of the environment does for the particular aspect considered (e.g. channel, etc.)